src/app/clusters/webrtc-transport-provider-server/webrtc-transport-provider-server.h

```diff
@@ -0,0 +1,225 @@
+/*
+ *
+ *    Copyright (c) 2025 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#pragma once
+
+#include <app-common/zap-generated/cluster-objects.h>
+#include <app/AttributeAccessInterface.h>
+#include <app/CommandHandlerInterface.h>
+#include <string>
+#include <vector>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace WebRTCTransportProvider {
+
+using ICEServerDecodableTypeStruct = Structs::ICEServerStruct::DecodableType;
+using WebRTCSessionTypeStruct      = Structs::WebRTCSessionStruct::Type;
+
+/**
+ * @brief
+ *   A delegate interface for the WebRTC Transport Provider cluster, allowing
+ *   OEM or application-specific logic to be implemented outside of the cluster
+ *   server itself. The server calls these methods to do the "real work."
+ */
+class Delegate
+{
+public:
+    virtual ~Delegate() = default;
+
+    struct OfferRequestArgs
+    {
+        uint16_t id;
+        StreamUsageEnum streamUsage;
+        Optional<DataModel::Nullable<uint16_t>> videoStreamID;
+        Optional<DataModel::Nullable<uint16_t>> audioStreamID;
+        Optional<DataModel::DecodableList<ICEServerDecodableTypeStruct>> iceServers;
```

Comment:
If we are using string/vector in here anyway, we should strongly consider not passing around DecodableList: it's hard to use right, and we don't want to force that on API consumers unless really needed.